### PR TITLE
Contributions from ugent.be migration

### PR DESCRIPTION
**Pull request by zopyx**
- export portlets
- export portlets blacklist
- export review state
- code cleanup (autopep8)
- configurable export directory through `EXPORT_DIR` environment variable

### CODE REVIEW

**Review by thet — APPROVED** [review]
LGTM

File: collective/jsonify/browser.py [hunk 1]
```diff
@@ -122,5 +122,5 @@ def push_json(self, objs):
             JSON = json.dumps(objs)
             self.request.response.setHeader("Content-type", "application/json")
             return JSON
-        except Exception, e:
+        except Exception as e:
```
Comment (pbauer) [inline collective/jsonify/browser.py:125]: This breaks Python 2.4 support

File: collective/jsonify/export.py [hunk 2]
```diff
@@ -295,6 +306,8 @@ def walk(folder, skip_callback=lambda item: False):
     :rtype: Plone context.
 
     """
+    print(folder)
+
```
Comment (pbauer) [inline collective/jsonify/export.py:310]: Please remove debug-print

File: collective/jsonify/export.py [hunk 3]
```diff
@@ -361,7 +374,7 @@ def write(items):
 
         try:
             context_dict = Wrapper(item)
-        except Exception, e:
+        except Exception as e:
```
Comment (pbauer) [inline collective/jsonify/export.py:377]: This breaks Python 2.4 support

File: collective/jsonify/export.py [hunk 4]
```diff
@@ -371,12 +384,13 @@ def write(items):
         while not passed:
             try:
                 # see, if we can serialize to json
-                json_structure = json.dumps(context_dict)  # noqa
+                json_structure = json.dumps(context_dict, cls=MyJSONEncoder)  # noqa
                 passed = True
-            except Exception, error:
+            except Exception as error:
```
Comment (pbauer) [inline collective/jsonify/export.py:389]: This breaks Python 2.4 support

File: collective/jsonify/methods.py [hunk 5]
```diff
@@ -35,7 +43,7 @@ def get_item(self):
 
     try:
         context_dict = Wrapper(self)
-    except Exception, e:
+    except Exception as e:
```
Comment (pbauer) [inline collective/jsonify/methods.py:46]: see above

File: collective/jsonify/methods.py [hunk 6]
```diff
@@ -47,7 +55,7 @@ def get_item(self):
         try:
             JSON = json.dumps(context_dict)
             passed = True
-        except Exception, error:
+        except Exception as error:
```
Comment (pbauer) [inline collective/jsonify/methods.py:58]: see above

File: collective/jsonify/wrapper.py [hunk 7]
```diff
@@ -911,9 +937,101 @@ def get_redirects(self):
...
+                                      for r in redirects]
         except:  # noqa: E722
             pass
+
+    def get_portlets(self):
+        """ Get portlet assignments """
+
+        from zope.component import getUtility
+        from zope.component import getMultiAdapter
+        from plone.portlets.interfaces import IPortletManager
```
Comment (pbauer) [inline collective/jsonify/wrapper.py:954]: I think adding imports from plone.portlets breaks support for plone versions older than 3.0

**Review by pbauer — CHANGES_REQUESTED** [review]

**Pull request closed without merging**